flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -97,6 +101,16 @@ public BlobServer(Configuration config) throws IOException {
 		this.storageDir = BlobUtils.initStorageDirectory(storageDirectory);
 		LOG.info("Created BLOB server storage directory {}", storageDir);
 
+		if(config.getBoolean(ConfigConstants.SECURITY_ENABLED, DEFAULT_SECURITY_ENABLED) == true) {
+			secureCookie = config.getString(ConfigConstants.SECURITY_COOKIE, null);
+			if(secureCookie == null) {
+				String errorMessage = "Missing " + ConfigConstants.SECURITY_COOKIE
+						+ " configuration in Flink config file";
+				LOG.error(errorMessage);
+				throw new IllegalConfigurationException(errorMessage);
+			}
+		}
+
```

Comment:
This looks a lot like duplicate code. Maybe it makes sense to move it to the `BlobUtils`, or even to the `SecurityUtils` as a static method, ensuring that security is enabled.